File: CombatTracker/CombatTracker.Web/ClientApp/app/entities/classes/EnumDefinitions.ts



    export module EnumDefinitions {

            export enum GameType {
                RMSS = 0,
                DnD5e = 1
            }
    
            export enum CriticalLevels {
                Same = 0,
                A = 1,
                B = 2,
                C = 3,
                D = 4,
                E = 5
            }
    
            export enum CharacterType {
                NPC = 0,
                PC = 1
            }
    
            export enum ActorActionType {
                None = 0,
                Normal = 1,
                Attack = 2,
                Spell = 3,
                Prep = 4,
                Movement = 5,
                Bleed = 6,
                Critical = 7,
                Death = 8,
                Psychic = 9,
                SpellEffect = 10
            }
    
            export enum ActionProblem {
                IsOK = 0,
                IsWarning = 1,
                IsError = 2
            }
    
            export enum ActionTypeEnum {
                None = 0,
                Current = 1,
                Proposed = 2,
                Next = 3,
                Effect = 4
            }
    
            export enum BaseActionType {
                None = 0,
                Normal = 1,
                Attack = 2,
                Spell = 3,
                Prep = 4,
                Movement = 5
            }
    
            export enum ParryType {
                Fine = 0,
                Must_Parry = 1,
                No_Parry = 2
            }
    
    }

File: CombatTracker/CombatTracker.Entities/Reference/EnumDefinitions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CombatTracker.Entities.Reference
{
    public enum ViewTypeEnum
    {
        Overview,
        Player,
        GM
    }


    public enum ResponseEnum
    {
        ChooseNextAction,
        DisplayMessage,
        AutoAdvance
    }

    public enum GameType
    {
        RMSS,
        DnD5e
    }

    public enum DamageLevel
    {
        Healthy,
        Winded,
        Bloodied,
        Rough,
        Unconscious,
        Dead
    }



    public enum ActorActionType
    {
        None,
        Normal,
        Attack,
        Spell,
        Prep,
        Movement,
        Bleed,
        Critical,
        Death,
        Psychic,
        SpellEffect
    }

    public enum ActionProblem
    {
        IsOK,
        IsWarning,
        IsError
    }

    public enum ActionTypeEnum
    {
        None,
        Current,
        Proposed,
        Next,
        Effect,
        Interrupted,
        Completed
    }

    public enum BaseActionType
    {
        None,
        Normal,
        Attack,
        Spell,
        Prep,
        Movement
    }

    public enum ParryType
    {
        Fine,
        Must_Parry,
        No_Parry
    }


    public enum CreatureStatus
    {
        Personal,
        Proposed,
        InCompendium
    }


    public enum CriticalLevels
    {
        Same,
        A,
        B,
        C,
        D,
        E
    }
}
